### PR TITLE
Sorting Resources

### PR DESCRIPTION
**Pull request by lovedeep92**
Nov-04-2020 21-38-00
**JIRA:**
See ticket here - https://jira.rax.io/browse/MNRVA-608

**Description:**

Resources page has the ability to sort.

Added sorting by resource id.

**Testing:**

`npm run test:unit`

 ## Screenshots:
(if applicable)

### CODE REVIEW

File: src/app/_features/resources/components/list/resourceslist.component.spec.ts [hunk 1]
```diff
@@ -294,21 +296,17 @@ it('should destroy subscriptions', (done) => {
...
     component.triggerConfirm();
-    expect(spy).toHaveBeenCalledTimes(3);
+    expect(spy).toHaveBeenCalledTimes(54);
   });
 
-  it('should execute delete multiple resources failed', () => {
-    component.selectedResources = [
-      {resourceId: "test-1", labels: {}, metadata: {}, presenceMonitoringEnabled: false, createdTimestamp: "2020-09-24T13:44:28Z",updatedTimestamp: "2020-09-24T13:44:28Z"},{resourceId: "test-2", labels: {}, metadata: {}, presenceMonitoringEnabled: false, createdTimestamp: "2020-09-24T13:44:43Z", updatedTimestamp:"2020-09-24T13:44:43Z"}];
-
+  fit('should execute delete multiple resources failed', () => {
```
Comment (FarajiA) [inline src/app/_features/resources/components/list/resourceslist.component.spec.ts:305]: Remove _fit_

File: src/app/_features/resources/components/list/resourceslist.component.spec.ts [hunk 2]
```diff
@@ -346,6 +356,11 @@ it('should destroy subscriptions', (done) => {
     let spy = spyOn(logService, 'log');
     component.triggerOk();
     expect(spy).toHaveBeenCalled();
-  })
+  });
+
+  fit('should check sorting resources desc', () => {
```
Comment (FarajiA) [inline src/app/_features/resources/components/list/resourceslist.component.spec.ts:361]: _fit_ here

File: src/app/_features/resources/components/list/resourceslist.component.ts [hunk 3]
```diff
@@ -57,10 +59,11 @@ export class ResourcesListComponent implements OnInit, OnDestroy {
 
   ngOnInit() {
     this.fetchResources = () => {
-      return this.resourceService.getResources(this.defaultAmount, this.page)
+      return this.resourceService.getResources(this.defaultAmount, this.page, this.sorting)
         .pipe(
           takeUntil(this.ngUnsubscribe)
         ).subscribe(() => {
+          console.log("this.resourceService.resources.content ", JSON.stringify(this.resourceService.resources.content));
```
Comment (FarajiA) [inline src/app/_features/resources/components/list/resourceslist.component.ts:66]: We can remove this log

File: src/app/_features/resources/components/list/resourceslist.component.html [hunk 4]
```diff
@@ -95,7 +95,16 @@ <h3>Add Resource</h3>
               </label>
             </hx-checkbox-control>
       </th>
-      <th>RESOURCE NAME</th>
+      <th>RESOURCE NAME
+        <button class="buttonIcon"  (click)="isDescending ? sortResources('desc', 'resourceId') : sortResources('asc', 'resourceId')">
+          <hx-icon *ngIf="isDescending"
```
Comment (FarajiA) [inline src/app/_features/resources/components/list/resourceslist.component.html:100]: Can we make these `<hx-icon>` elements have the hand pointer when hovered over? CSS would be something like `cursor: pointer`.

File: src/app/_features/resources/components/list/resourceslist.component.ts [hunk 5]
```diff
@@ -95,6 +98,14 @@ export class ResourcesListComponent implements OnInit, OnDestroy {
       e["checked"] = event.target.checked;
     });
   }
+
+  sortResources(orderBy, sortBy) {
```
Comment (FarajiA) [inline src/app/_features/resources/components/list/resourceslist.component.ts:102]: Since this has params and uses other functions let's add comments.

**Review by FarajiA — CHANGES_REQUESTED** [review]

File: src/app/_features/resources/components/list/resourceslist.component.html (hunk 4 above)
Comment (lovedeep92) [reply src/app/_features/resources/components/list/resourceslist.component.html:100]: done !!

File: src/app/_features/resources/components/list/resourceslist.component.ts (hunk 3 above)
Comment (lovedeep92) [reply src/app/_features/resources/components/list/resourceslist.component.ts:66]: done!!

File: src/app/_features/resources/components/list/resourceslist.component.spec.ts (hunk 2 above)
Comment (lovedeep92) [reply src/app/_features/resources/components/list/resourceslist.component.spec.ts:361]: done!!

File: src/app/_features/resources/components/list/resourceslist.component.spec.ts (hunk 1 above)
Comment (lovedeep92) [reply src/app/_features/resources/components/list/resourceslist.component.spec.ts:305]: done!!

File: src/app/_features/resources/components/list/resourceslist.component.ts (hunk 5 above)
Comment (lovedeep92) [reply src/app/_features/resources/components/list/resourceslist.component.ts:102]: done!!

**Review by FarajiA — APPROVED** [review]

**Pull request merged**